src/checkers/firefoxchecker.py

```
@@ -0,0 +1,181 @@
+            return None
+
+        added = []
+        processed_data = []
+        for data in external_data:
+            if data.filename in latest_info:
+                new_version = latest_info[data.filename]
+                data.state = ExternalData.State.VALID
+                if not data.current_version.matches(new_version):
+                    data.new_version = new_version
+            else:
+                if re.match(checker_data['browser']['filename-match'], data.filename) or \
+                   re.match(checker_data['translations']['filename-match'], data.filename):
+                    data.state = ExternalData.State.REMOVED
+            processed_data.append(data.filename)
+
+        added = []
+        new_sources = []
+        for source_filename, info in latest_info.items():
+            if not source_filename in processed_data:
+                source = {
+                    'filename': source_filename,
+                    'type': 'extra-data',
+                    'url': info.url,
+                    'sha256': info.checksum,
+                    'size': info.size,
+                }
+                # add new data to the same manifest as the main browser tarball
+                data = ExternalDataSource.from_source(browser_data.source_path, source, browser_data.source_parent)
+                data.state = ExternalData.State.ADDED
+                added.append(data)
+
+        return added
+
+    def _get_latest_available_version(self):
+        if self._latest_version:
+            return
+
+        latest_version_data = {}
+        request_url = self.FIREFOX_RELEASE_INFO_URL
+        with urllib.request.urlopen(request_url) as resp:
+            latest_version_data = json.load(resp)
+
+        if isinstance(latest_version_data, dict) and 'LATEST_FIREFOX_VERSION' in latest_version_data:
+            self._latest_version = latest_version_data['LATEST_FIREFOX_VERSION']
+        else:
+            self._latest_version = None
```

We should fail in this case.

---

This method is invoked inside a `try/catch/finally` and the `self._latest_version` is checked on `finally` with a warning if set to `None`.

Do you think we should raise an exception (i.e. break the checker) in this case? Or how else 
 would you fail here?

---

> Do you think we should raise an exception (i.e. break the checker) in this case?

Probably! If this web service turns out to be unreliable we can do something different.

---

Done.